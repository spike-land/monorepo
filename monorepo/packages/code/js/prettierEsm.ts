import Prettier from "prettier/esm/standalone.mjs";
import parserBabel from "prettier/esm/parser-babel.mjs";
import parserHtml from "prettier/esm/parser-html.mjs";
import parserPostcss from "prettier/esm/parser-postcss.mjs";


const jsOptions = {
  arrowParens: "always",
  bracketSpacing: true,
  embeddedLanguageFormatting: "auto",
  insertPragma: false,
  bracketSameLine: true,
  jsxSingleQuote: false,
  htmlWhitespaceSensitivity: "strict",
  printWidth: 90,
  proseWrap: "preserve",
  quoteProps: "as-needed",
  requirePragma: false,
  semi: true,
  singleQuote: true,
  tabWidth: 2,
  trailingComma: "all",
  useTabs: false,
  parser:  "babel-ts",
  plugins: [parserBabel, parserHtml, parserPostcss]
}

const htmlOptions ={
  htmlWhitespaceSensitivity: "css",
  parser: "html",
  plugins: [parserHtml, parserPostcss]
}

const cssOptions = {
  parser:  "css",
  plugins: [ parserPostcss]
}

export const prettier = (code: string) =>Prettier.format(code,jsOptions);
export const prettierCss = (code: string) =>Prettier.format(code,cssOptions);
export const prettierHtml = (code: string) =>Prettier.format(code,htmlOptions);

