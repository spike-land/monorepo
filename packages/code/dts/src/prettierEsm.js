import pluginEstree from "prettier/plugins/estree";
import pluginTypescript from "prettier/plugins/typescript";
import { format } from "prettier/standalone";
export const prettierJs = async (code) => {
  try {
    return await format(code, {
      arrowParens: "always",
      bracketSpacing: true,
      embeddedLanguageFormatting: "auto",
      insertPragma: false,
      bracketSameLine: true,
      jsxSingleQuote: false,
      htmlWhitespaceSensitivity: "strict",
      printWidth: 90,
      proseWrap: "preserve",
      quoteProps: "as-needed",
      requirePragma: false,
      semi: true,
      singleQuote: false,
      tabWidth: 2,
      trailingComma: "all",
      useTabs: false,
      parser: "typescript",
      plugins: [pluginEstree, pluginTypescript],
    });
  } catch (error) {
    if (code === "Types not found") {
      return "export {}";
    }
    console.error("prettier error"), console.error({ err: error, code });
    return code;
  }
};
Object.assign(self, { prettierJs });
