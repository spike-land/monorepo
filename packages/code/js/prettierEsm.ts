import Prettier from "prettier/standalone";
import parserBabel from "prettier/parser-babel";
import parserHtml from "prettier/parser-html";
import parserPostcss from "prettier/parser-postcss";

export const prettierJs = (code: string) =>
  Prettier.format(code, {
    arrowParens: "always",
    bracketSpacing: true,
    embeddedLanguageFormatting: "auto",
    insertPragma: false,
    bracketSameLine: true,
    jsxSingleQuote: false,
    htmlWhitespaceSensitivity: "strict",
    printWidth: 90,
    proseWrap: "preserve",
    quoteProps: "as-needed",
    requirePragma: false,
    semi: true,
    singleQuote: true,
    tabWidth: 2,
    trailingComma: "all",
    useTabs: false,
    parser: "babel-ts",
    plugins: [parserBabel, parserHtml, parserPostcss],
  });

export const prettierCss = (code: string) =>
  Prettier.format(code, {
    parser: "css",
    plugins: [parserPostcss],
  }) as string;
export const prettierHtml = (code: string) =>
  Prettier.format(code, {
    htmlWhitespaceSensitivity: "css",
    parser: "html",
    plugins: [parserHtml, parserPostcss],
  }) as string;
