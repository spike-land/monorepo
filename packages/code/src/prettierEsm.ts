// import parserBabel from "prettier/parser-babel";
import pluginTypescript from "prettier/plugins/typescript";
// @ts-ignore
import pluginEstree from "prettier/plugins/estree";
import { format } from "prettier/standalone";
// export type * as Prettier from "prettier/standalone"
// import parserHtml from "prettier/parser-html";
// import parserPostcss from "prettier/parser-postcss";

// let lastSuccessful: string | null = null;

export const prettierJs = async (code: string, toThrow = false) => {
  // return code;
  // console.log(`prettierJS: ${code}`);
  try {
    return await format(code, {
      arrowParens: "always",
      bracketSpacing: true,
      embeddedLanguageFormatting: "auto",
      insertPragma: false,
      // bracketSpacing: true,

      bracketSameLine: true,
      jsxSingleQuote: false,

      htmlWhitespaceSensitivity: "strict",
      printWidth: 90,
      proseWrap: "preserve",
      quoteProps: "as-needed",
      requirePragma: false,
      semi: true,

      singleQuote: false,

      tabWidth: 2,
      trailingComma: "all",
      useTabs: false,
      parser: "typescript",
      plugins: [pluginEstree, pluginTypescript /// parserHtml, parserPostcss
      ],
    });
    // return lastSuccessful = current;
  } catch (error) {
    if (toThrow) throw error;
    if (code === "Types not found") return "export {}";
    console.error("prettier error"), console.error({ err: error, code });

    return code;
  }
};

Object.assign(self, { prettierJs });

// export const prettierCss = (code: string) =>
//   Prettier.format(code, {
//     parser: "css",
//     plugins: [parserPostcss],
//   });
// export const prettierHtml = (code: string) =>
//   Prettier.format(code, {
//     htmlWhitespaceSensitivity: "css",
//     parser: "html",
//     plugins: [parserHtml, parserPostcss],
//   });
