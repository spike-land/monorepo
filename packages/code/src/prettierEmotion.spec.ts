import { describe, it, expect } from 'vitest';
import { addSomeFixesIfNeeded } from "./prettierEsm";

describe("addSomeFixesIfNeeded", () => {
  it("adds default export if 'export default' is missing and 'const App' is present", () => {
    const code = "const App = () => {};";
    expect(addSomeFixesIfNeeded(code)).toMatchSnapshot();
  });

  it("adds import statement for '@emotion/react' if 'css={css`' is present and '@emotion/react' is missing", () => {
    const code = "css={css`color: red;`}";
    expect(addSomeFixesIfNeeded(code)).toMatchSnapshot();
  });

  it("adds import statement for 'FC' if ' FC<{' is present and 'import type { FC }' is missing", () => {
    const code = "FC<{ prop: string }>";
    expect(addSomeFixesIfNeeded(code)).toMatchSnapshot();
  });

  it("adds import statement for 'useState' if 'useState(' is present and 'import { useState }' is missing", () => {
    const code = "useState(0)";
    expect(addSomeFixesIfNeeded(code)).toMatchSnapshot();
  });

  it("handles multiple occurrences of 'css={css`' correctly", () => {
    const code = "css={css`color: red;`} css={css`font-size: 16px;`}";
    expect(addSomeFixesIfNeeded(code)).toMatchSnapshot();
  });

  it("doesnt don stupid things", () => {
    const code = `
    <motion.pre
      className="absolute bottom-24 left-24 font-mono text-xs text-yellow-400"
      initial={{ y: 100, opacity: 0 }}
      animate={{ y: 0, opacity: 1 }}
      transition={{ duration: 1 }}
      css={css\`
        white-space: pre;
      \`}
    >
      <motion.div className="absolute inset-0 pointer-events-none">
        {[...Array(10)].map((_, i) => (
          <motion.div
            key={i}
            className="absolute"
            style={{
              top: \`${Math.random() * 100}%\`,
              left: \`${Math.random() * 100}%\`,
              width: "4px",
              height: "4px",
              borderRadius: "50%",
              backgroundColor: ["#ff0000", "#00ff00", "#0000ff", "#ffff00", "#ff00ff"][
                Math.floor(Math.random() * 5)
              ],
            }}
            variants={fireworkVariants}
            initial="initial"
            animate="animate"
            transition={{ repeat: Infinity, repeatDelay: Math.random() * 3 + 1 }}
          />
        ))}
      </motion.div>
    </motion.pre>
  `;
    expect(addSomeFixesIfNeeded(code)).toMatchSnapshot();
  });
});