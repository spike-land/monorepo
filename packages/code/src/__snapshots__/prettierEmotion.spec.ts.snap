// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`addSomeFixesIfNeeded adds default export if 'export default' is missing and 'const App' is present 1`] = `"const App = () => {};"`;

exports[`addSomeFixesIfNeeded adds import statement for '@emotion/react' if 'css={css\`' is present and '@emotion/react' is missing 1`] = `
"css={css\`
    color: red;
  \`}"
`;

exports[`addSomeFixesIfNeeded adds import statement for 'FC' if ' FC<{' is present and 'import type { FC }' is missing 1`] = `"FC<{ prop: string }>"`;

exports[`addSomeFixesIfNeeded adds import statement for 'useState' if 'useState(' is present and 'import { useState }' is missing 1`] = `"useState(0)"`;

exports[`addSomeFixesIfNeeded handles multiple occurrences of 'css={css\`' correctly 1`] = `
"css={css\`
    color: red;
  \`} css={css\`
    font-size: 16px;
  \`}"
`;
